velociraptor/autoplotter/compare.py

```diff
@@ -164,15 +164,26 @@ def recreate_single_figure(
     for line_type in valid_line_types:
         line = getattr(plot, f"{line_type}_line", None)
         if line is not None:
-            for name, data in line_data.items():
+            for color, (name, data) in enumerate(line_data.items()):
+                color = f"C{color}"
+
+                try:
+                    this_line_dict = data[plot.filename]["lines"][line_type]
+                    centers = unyt.unyt_array(
+                        this_line_dict["centers"], units=plot.x_units
+                    )
+                    heights = unyt.unyt_array(
+                        this_line_dict["values"], units=plot.y_units
+                    )
+                    errors = unyt.unyt_array(
+                        this_line_dict["scatter"], units=plot.y_units
+                    )
+                except KeyError:
+                    continue
+
                 ax.set_xlabel(data[plot.filename].get("x_label"))
```

Comment:
```suggestion
                if data[plot.filename].get("x_label"):
                    ax.set_xlabel(data[plot.filename].get("x_label"))
```

Reply:
This will not let `data[plot.filename].get("x_label")=None` propagate to figure's axis labels.

Reply:
That should be handled by the `continue` above

Reply:
As far as I remember,

```
except KeyError:
                    continue
```

didn't work, because although equal to **None**, the attribute "x_label" was still defined.

I will double-check it.

Reply:
The KeyError will be generated when we do `data[plot.filename]`